crates/rpc/src/handlers/validator.rs

```diff
@@ -348,3 +348,68 @@ pub async fn post_validator_balances_from_state(
         ))),
     )
 }
+
+#[derive(Debug, Serialize)]
+pub struct ValidatorLivenessData {
+    #[serde(with = "serde_utils::quoted_u64")]
+    pub index: u64,
+    pub is_live: bool,
+}
+
+impl ValidatorLivenessData {
+    pub fn new(index: u64, is_live: bool) -> Self {
+        Self { index, is_live }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct ValidatorIndex(#[serde(with = "serde_utils::quoted_u64")] pub u64);
+
+impl Deref for ValidatorIndex {
+    type Target = u64;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+#[post("/validator/liveness/{epoch}")]
+pub async fn post_validator_liveness(
+    db: Data<ReamDB>,
+    epoch: Path<u64>,
+    validator_indices: Json<Vec<ValidatorIndex>>,
+) -> Result<impl Responder, ApiError> {
+    let epoch = epoch.into_inner();
+    let validator_indices = validator_indices.into_inner();
+
+    let state = get_state_from_id(ID::Slot(epoch * SLOTS_PER_EPOCH), &db).await?;
+
+    let mut liveness_data = Vec::new();
+
+    for validator_index in validator_indices {
```

Comment:
```suggestion
    let state = get_state_from_id(ID::Slot(epoch * SLOTS_PER_EPOCH), &db).await?;

    let mut liveness_data = Vec::new();

    for validator_index in validator_indices.into_inner() {
```

you might as well inline this, if you also need to fix CI